doc/md: add instructions for migration dir linting in ci page (#2865)
ent/ent: doc/md/ci.mdx

</TabItem>
</Tabs>

## Lint migration files

Changes to your project's Ent schema almost always result in a modification
of your database. If you are using [Versioned Migrations](/docs/versioned-migrations)
to manage changes to your database schema, we recommend running [migration linting](https://atlasgo.io/versioned/lint)
as part of your continuous integration flow. This is done for multiple reasons:

* Linting replays your migration directory on a real database to make sure all SQL statements
  are valid and in the correct order.
* Directory integrity is enforced - ensuring that history wasn't accidentally changed and

>   that migrations that are planned in parallel are unified to a clean linear history.

Reviewer: Link to https://entgo.io/docs/versioned-migrations#atlas-migration-directory-integrity-file

* Destructive changes are detected notifying you of any potential data loss that may be
  caused by your migrations way before they reach your production database.
* Linting detects data-dependant changes that _may_ fail upon deployment and require
  more careful review from your side.

If you're using GitHub, we recommend using the [Official Atlas Action](https://github.com/ariga/atlas-action)
to run migration linting during CI.

Add `.github/workflows/atlas-ci.yaml` to your repo with the following contents:
<Tabs
defaultValue="mysql"
values={[
    {label: 'MySQL', value: 'mysql'},
    {label: 'MariaDB', value: 'maria'},
    {label: 'PostgreSQL', value: 'postgres'},
]}>
<TabItem value="mysql">

```yaml
name: Atlas CI
on:
  # Run whenever code is changed in the master branch,
  # change this to your root branch.
  push:
    branches:
      - master
  # Run on PRs where something changed under the `ent/migrate/migrations/` directory.
  pull_request:
    paths:
      - 'ent/migrate/migrations/*'
jobs:
  lint:
    services:
      # Spin up a mysql:8.0.29 container to be used as the dev-database for analysis.

>       mysql:

Reviewer: This service is called `mysql` but maria and postgres are called with their version `postgres10` / `maria107`

        image: mysql:8.0.29
        env:
          MYSQL_ROOT_PASSWORD: pass
          MYSQL_DATABASE: test
        ports:

>           - "3307:3306"

Reviewer: You are quoting the ports here but not on maria and postgres

          dir-format: golang-migrate # Or: atlas, goose, dbmate
          dev-url: mysql://root:pass@localhost:3307/test
```

</TabItem>
<TabItem value="maria">

```yaml
name: Atlas CI
on:
  # Run whenever code is changed in the master branch,
  # change this to your root branch.
  push:
    branches:
      - master
  # Run on PRs where something changed under the `ent/migrate/migrations/` directory.
  pull_request:
    paths:
      - 'ent/migrate/migrations/*'
jobs:
  lint:
    services:
      # Spin up a maria:10.7 container to be used as the dev-database for analysis.
      maria107:
        image: mariadb:10.7
        env:
          MYSQL_DATABASE: test
          MYSQL_ROOT_PASSWORD: pass
        ports:
          - 4306:3306
        options: >-
          --health-cmd "mysqladmin ping -ppass"
          --health-interval 10s
          --health-start-period 10s
          --health-timeout 5s
          --health-retries 10
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3.0.1
        with:
          fetch-depth: 0 # Mandatory unless "latest" is set below.
      - uses: ariga/atlas-action@v0
        with:
          dir: path/to/migrations

>           dir-format: golang-migrate # Or: atlas, goose, dbmate

Reviewer: We have flyway and liquibase as well.

Also we already depcrecated golang migrate as default. I think we can use `atlas` here already.

</TabItem>
</Tabs>

## Lint migration files

Changes to your project's Ent schema almost always result in a modification
of your database. If you are using [Versioned Migrations](/docs/versioned-migrations)
to manage changes to your database schema, we recommend running [migration linting](https://atlasgo.io/versioned/lint)
as part of your continuous integration flow. This is done for multiple reasons:

> * Linting replays your migration directory on a real database to make sure all SQL statements

Reviewer 2: what do you mean by real? let's call it a "container" and link to the Atlas dev-database doc.

  are valid and in the correct order.
* [Migration directory integrity](/docs/versioned-migrations#atlas-migration-directory-integrity-file)
  is enforced - ensuring that history wasn't accidentally changed and that migrations that are
  planned in parallel are unified to a clean linear history.
* Destructive changes are detected notifying you of any potential data loss that may be
  caused by your migrations way before they reach your production database.
* Linting detects data-dependant changes that _may_ fail upon deployment and require
  more careful review from your side.

> If you're using GitHub, we recommend using the [Official Atlas Action](https://github.com/ariga/atlas-action)

Reviewer 2: ```suggestion
If you're using GitHub, you can use the [Atlas Action](https://github.com/ariga/atlas-action)
```
